src/VisualStudio/IntegrationTest/IntegrationTests/VisualBasic/BasicFormatting.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.VisualStudio.IntegrationTest.Utilities;
+using Microsoft.VisualStudio.IntegrationTest.Utilities.Input;
+using Roslyn.Test.Utilities;
+using Roslyn.VisualStudio.IntegrationTests.Extensions;
+using Roslyn.VisualStudio.IntegrationTests.Extensions.Editor;
+using Xunit;
+
+// *************************************************************
+// ***** Turn on visible whitespace when editing this file *****
+// *************************************************************
+
+namespace Roslyn.VisualStudio.IntegrationTests.VisualBasic
+{
+    [Collection(nameof(SharedIntegrationHostFixture))]
+    public class BasicFormatting : AbstractEditorTest
+    {
+        protected override string LanguageName => LanguageNames.VisualBasic;
+
+        public BasicFormatting(VisualStudioInstanceFactory instanceFactory)
+            : base(instanceFactory, nameof(BasicFormatting))
+        {
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Formatting)]
+        public void VerifyFormattingIndent()
+        {
+            SetUpEditor(@"
+$$Module A
+    Sub Main(args As String())
+    
+        End Sub
+End Module");
+            Editor.FormatDocument();
+            this.VerifyTextContains(@"
+Module A
+    Sub Main(args As String())
+
+    End Sub
+End Module");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Formatting)]
+        public void VerifyCaseCorrection()
+        {
+            SetUpEditor(@"
+$$module A
+end module");
+            Editor.FormatDocument();
+            this.VerifyTextContains(@"
+Module A
+End Module");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Formatting)]
+        public void ShiftEnterWithIntelliSenseAndBraceMatching()
```

Comment:
I added the equivalent one for C#, but marked them both as Skipped, since it doesn't work correctly in either language.